### PR TITLE
Add stub and test to lofreq/indelqual

### LINKED ISSUE #5940: [FEATURE] add test and stub to lofreq/indelqual
### Is your feature request related to a problem? Please describe

The lofreq/indelqual has no tests or stub yet.

### Describe the solution you'd like

Adding to the existing module a stub output and a test

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Krannich479**
## PR checklist

Closes #5940 

This PR commits a new stub and test for lofreq/indelqual.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/lofreq/indelqual/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
+                    ]
+                input[1] = [
+                        file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+                    ]
```
Comment (famosab) [inline modules/nf-core/lofreq/indelqual/tests/main.nf.test:23]:
```suggestion
                input[0] = [  [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
                    ]
                input[1] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
```
Can you update the paths as described here: https://nf-co.re/docs/guidelines/components/modules#input-data

File: modules/nf-core/lofreq/indelqual/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
+    test("sarscov2 - bam, fasta - stub") {
```
Comment (famosab) [inline modules/nf-core/lofreq/indelqual/tests/main.nf.test:37]: You wont need to add the fasta file to the test name, here the "main" input is enough IMO

**Review by famosab — COMMENTED** [review]
Did you use 
`nf-core modules create lofreq/indelqual --migrate-pytest`
to create the files? If not, check it out and if yes: You should still push that you deleted the pytest files 👍

**Comment by Krannich479** [comment]
> Did you use `nf-core modules create lofreq/indelqual --migrate-pytest` to create the files? If not, check it out and if yes: You should still push that you deleted the pytest files 👍

Did this for the existing module. Added an updated test file plus the file deletions after your recommended command above.

**Comment by Krannich479** [comment]
The snapshot's md5 sum for the output BAM is different for docker and singularity. Is this a known issue?

**Comment by Krannich479** [comment]
I looked into what @MarieLataretu did for other `lofreq` modules. 
#5420 : the fix was to add new asserts, checking existence of the BAM + match of versions log
#5197 : the fix was to add the samtools version to the `environment.yml`

---

As I think #5197 is more bulletproof since it keeps checking BAM file consistency. So I tried this. However, neither `samtools` v1.12 (as in the container of this module) nor v1.17 (as in #5197 ) worked in my case. The test always failed with a distinct md5 sum from singularity. I won't test arbitrarily many more samtools versions.
Hence, I implemented the tests of #5420 for this module too.

File: modules/nf-core/lofreq/indelqual/meta.yml [hunk 3]
```diff
@@ -20,10 +22,15 @@ input:
...
+  - meta2:
+      type: map
+      description: |
+        Groovy Map containing sample information about the reference fasta
+        e.g. [ id:'reference' ]
   - fasta:
       type: file
       description: Reference genome FASTA file
-      pattern: "*.{fasta}"
+      pattern: "*.{fasta/fa}"
```
Comment (vagkaratzas) [inline modules/nf-core/lofreq/indelqual/meta.yml:33]: Should be fasta,fa with a comma

File: modules/nf-core/lofreq/indelqual/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                {
+                    def bam = file(process.out.bam.get(0).get(1))
+                    assert bam.exists()
+                },
+                { assert path(process.out.versions.get(0)).md5 == "822d596299afbb3ae6db66bfa9b77b3e" }
```
Comment (vagkaratzas) [inline modules/nf-core/lofreq/indelqual/tests/main.nf.test:69]: At least for the stub test it should be as simple as:
{ assert snapshot(process.out).match() }

File: modules/nf-core/lofreq/indelqual/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
```
Comment (vagkaratzas) [inline modules/nf-core/lofreq/indelqual/tests/main.nf.test:37]: Better like this maybe: { assert snapshot(process.out.versions).match("versions") } ? To keep the same across all modules

File: modules/nf-core/lofreq/indelqual/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                {
+                    def bam = file(process.out.bam.get(0).get(1))
+                    assert bam.exists()
```
Comment (vagkaratzas) [inline modules/nf-core/lofreq/indelqual/tests/main.nf.test:35]: If assert snapshot(process.out).match() does not work, then probably check if a certain line exists in the file. Then if for some reason that's not possible, fallback to the .exists() check.

**Review by vagkaratzas — CHANGES_REQUESTED** [review]
Good stuff! Some minor points to address.

File: modules/nf-core/lofreq/indelqual/tests/main.nf.test (hunk 6 above)
Comment (Krannich479) [reply modules/nf-core/lofreq/indelqual/tests/main.nf.test:35]: In the past, I have struggled to get the string comparison of file content working. But I agree and will give it another shot. 👍

File: modules/nf-core/lofreq/indelqual/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                },
```
Comment (vagkaratzas) [inline modules/nf-core/lofreq/indelqual/tests/main.nf.test:68]: Ok, just remove the above assertion from the stub test and it should be ok!

**Review by vagkaratzas — APPROVED** [review]
Minor comment and then feel free to merge

**Comment by Krannich479** [comment]
Thanks for your feedback!

**Pull request merged**